cube_helper/cube_loader.py

```diff
@@ -12,6 +12,52 @@
 from datetime import datetime
 
 
+def _fix_partial_datetime(constraint):
+    if isinstance(constraint._coord_values['time'], iris.time.PartialDateTime):
+        part_datetime = constraint._coord_values['time']
+        if part_datetime.year:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.year == part_datetime.year)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.month == part_datetime.month)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.day == part_datetime.day)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.hour == part_datetime.hour)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.minute == part_datetime.minute)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.second == part_datetime.second)
+        elif part_datetime.month:
+            new_constraint = iris.Constraint(
+                time=lambda cell:
+                cell.point.microsecond == part_datetime.microsecond)
+        else:
+            raise OSError("Constraint could not be rectified")
+        return new_constraint
+    else:
+        return constraint
+
+
+def _constraint_compatible(cube, constraint):
+    try:
+        constraint.extract(cube)
+        return True
+    except Exception:
```

Comment:
Rather than being very generic with `except Exception:` is it worth being more specific? I've just had a quick look and I've seen extract fail with TypeError. It would be really good if you could have a play to say what errors extract does raise and just catch these. You could even have a look at the Iris source code to see if it's obvious from there what exceptions it will raise.